server/src/main/java/org/opensearch/cluster/decommission/DecommissionController.java

```diff
@@ -267,4 +278,94 @@ public void clusterStateProcessed(String source, ClusterState oldState, ClusterS
             }
         });
     }
+
+    void setWeights(String awarenessAttribute, Map<String, Double> weights, ActionListener<ClusterPutWeightedRoutingResponse> listener) {
```

Comment:
Let me rethink of a name. We will just set the weights in this method. This method will be used for recommission as well. And we can pass the weights and keep the call to weights api generic. Hence i have split the methods to avoid code duplication.

Reply:
Updated to ```setRoutingWeights```